test/ValidationFramework/CrossValidator.cpp

```diff
@@ -0,0 +1,125 @@
+/// Apache License 2.0
+
+#include <ValidationFramework/CrossValidator.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace validation
+{
+
+Array<Array<VectorXd>> CrossValidator::IngestOutputQuantities(
+    const Table& aTable, const ToolComparison& aToolComparison
+)
+{
+    const Array<QuantityComparison> quantityComparisons = aToolComparison.quantityComparisons;
+
+    Array<Array<VectorXd>> groupedCoordinatesArray = Array<Array<VectorXd>>::Empty();
+
+    for (const auto& row : aTable)
+    {
+        Array<VectorXd> coordinateSubsets = Array<VectorXd>::Empty();
+
+        Size rowReadIndex = 0;
+        for (Size coordinateSubsetIndex = 0; coordinateSubsetIndex < quantityComparisons.getSize();
+             coordinateSubsetIndex++)
+        {
+            const Quantity quantity = quantityComparisons[coordinateSubsetIndex].quantity;
+            const Size quantityDimension = QuantityDimensionMap.at(quantity);
+
+            VectorXd coordinateSubset(quantityDimension);
+
+            for (Size quantityDimensionIndex = 0; quantityDimensionIndex < quantityDimension; quantityDimensionIndex++)
+            {
+                if ((aToolComparison.tool == Tool::GMAT) && (quantity != Quantity::MASS))
+                {
+                    coordinateSubset(quantityDimensionIndex) = row[1 + rowReadIndex].accessReal() * 1.0e3;
+                }
+                else
+                {
+                    coordinateSubset(quantityDimensionIndex) = row[1 + rowReadIndex].accessReal();
+                }
+                rowReadIndex++;
+            }
+            coordinateSubsets.add(coordinateSubset);
+        }
+        groupedCoordinatesArray.add(coordinateSubsets);
+    }
+
+    // TBI: read the "seconds elapsed" from reference data and check to make sure its within tolerance of ostk
+    // timestamps
+    return groupedCoordinatesArray;
+}
+
+VectorXd CrossValidator::CompareOutputQuantities(
+    const State& anOSTkState,
+    const Array<VectorXd>& aReferenceCoordinates,
+    const Array<QuantityComparison>& aQuantityComparisons
+)
+{
+    VectorXd errorsPerQuantities(aQuantityComparisons.getSize());
```

Comment:
I would switch `error` to `delta` because uncertainties are present everywhere including in other providers externally (nit)